src/client/src/shell/components/app-header/Header.tsx

```diff
@@ -5,16 +5,73 @@ import { styled, ThemeName, useTheme } from 'rt-theme'
 import Logo from './Logo'
 
 class Header extends React.Component {
+  // The added buttons handle the Spot windows management
+
+  // The state handles the buttons visibility
+  state = {
+    showStackAll: false,
+    showTabAll: false,
+    showToggleCollapse: false,
+  }
+
+  componentDidMount() {
+    // On Spot window open/close the buttons visibility is updated
+    const interval = setInterval(() => {
+      if (!(window as any).glue) {
+        return
+      }
+      (window as any).glue.agm.register(
```

Comment:
We cant have references to specific desktop frameworks such as Glue in the main application code outside of an extension point. Otherwise, our application would be littered with vendor-specific code. We will need a way to inject UI elements in strategic places like the application header. I'll have a think about this and get back to you.